[SDK] Fixed "Move to... & Copy to..." dialog broken button navigation

Gathub22 opened the pull request:
Fixes CORE-19585

## Purpose

I've added the corresponding patch that fixes the keyboard navigation bug that makes the cursor get stuck, according to [CORE-19585](https://jira.reactos.org/browse/CORE-19585). ~The patch just now checks for SBS_SIZEBOX and SBS_SIZEGRIP.~ After some research, this patch just adds WS_GROUP to layout.h to avoid getting the cursor stuck. 

With this path, the bug is fixed:
### "Move to..." menu
![2024-05-20 22-33-52](https://github.com/reactos/reactos/assets/117179675/1e39741d-8756-4473-8aed-7456e5f62a79)

### "Copy to..." menu
![imagen](https://github.com/reactos/reactos/assets/117179675/32fd83e8-6bad-4601-b95a-a4555a2728db)

HBelusca reviewed (commented):
Before approving the PR, I would invite you to test something:
Prepare a Windows XP/2003 virtual machine, where you also install a tool like e.g. [WinSpy](https://www.catch22.net/projects/winspy/) , that allows you to change at runtime the styles of different windows.

Then, open a window (dialog or something else?) that you know has a resizing grip control, and use WinSpy to change the styles of that control at runtime: for example, remove `SBS_SIZEBOX` and `SBS_SIZEGRIP`, and maybe enlarge the control size too.
After that, test whether, by pressing TAB to circulate through the list of dialog controls, the scrollbar gets selected at all.

Gathub22 commented:
> Before approving the PR, I would invite you to test something: Prepare a Windows XP/2003 virtual machine, where you also install a tool like e.g. [WinSpy](https://www.catch22.net/projects/winspy/) , that allows you to change at runtime the styles of different windows.
> 
> Then, open a window (dialog or something else?) that you know has a resizing grip control, and use WinSpy to change the styles of that control at runtime: for example, remove `SBS_SIZEBOX` and `SBS_SIZEGRIP`, and maybe enlarge the control size too. After that, test whether, by pressing TAB to circulate through the list of dialog controls, the scrollbar gets selected at all.
> 
> Basically, what I'm wondering is whether it's instead the scrollbar control who should ignore focus, whenever it is either a sizebox or grip. It would be here: https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/user32/controls/scrollbar.c;hb=43b3280f69693d7f3b011b17cf43831fbf6b63c7#l1308 that the change would have to be implemented in that case.
> 
> I also wonder whether this "skipping the focus" behaviour is a behaviour that is happening only when the sizegrip/box control is put on a dialog (but not on a regular window), or whether it happens anywhere. See for example the ReactOS eventvwr main window, that has such a control, but isn't a dialog. If the "skipping the focus" behaviour happens when the control is used _anywhere_, then this would indicate that the correct fix is instead, in the scrollbar.c file (as I pointed in the link above). If this happens _only_ when the control is in a dialog, but not elsewhere (in a non-dialog window), _then_, this would instead point towards this PR.
> 
> Besides all this discussion, I've just noticed this in the message handling of scrollbar when it has the grip/box style: https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/user32/controls/scrollbar.c;hb=43b3280f69693d7f3b011b17cf43831fbf6b63c7#l799
> 
> ```c
>  813           case WM_LBUTTONUP:
>  814             ReleaseCapture();
>  815             ScrollTrackHitTest = HitTest = SCROLL_NOWHERE;
>  816             if (Wnd == GetFocus()) ShowCaret(Wnd);
>  817             break;
> ```
> 
> I wonder then how this GetFocus thing would do what it's supposed to do, if we now make the control skip the focus.

I'll check that later :)

Gathub22 commented:
I've reverted the old patch since it wasn't the best to fix this. Instead I've added [the new one](https://jira.reactos.org/secure/attachment/68183/68183_WS_GROUP.patch).

Gathub22 commented:
And the bug remains fixed:

### Move to...
![2024-05-23 14-17-57](https://github.com/reactos/reactos/assets/117179675/772da6ba-d38d-4063-87cf-da960632d24b)

### Copy to...
![2024-05-23 14-49-30](https://github.com/reactos/reactos/assets/117179675/ef9e7bc6-7402-444b-8fcd-b48f46c13f17)

learn-more commented on win32ss/user/user32/windows/dialog.c:
```diff
@@ -2104,7 +2104,7 @@ EndDialog(
        owner = GetAncestor( hwnd, GA_PARENT);
     }
     else
-       owner = GetWindow( hwnd, GW_OWNER );    
```
Please revert all formatting changes in this file.

JoachimHenze reviewed (approved):
I do suspect that Mark prefers to avoid the fix to become a commit touching several modules at once unnecessarily. Yes, that makes sense. Your changes are good. He just wants to see them separated. Respecting his wish, I do give my approval here for the layout.h changes therefore.

HBelusca reviewed (approved):
OK. As Mark said, the formatting changes in user32/windows/dialog.c are unwanted for this PR (since nothing gets fixed in that file.)

cbialorucki reviewed (approved):
Please undo all formatting/whitespace changes before merging.

binarymaster commented on sdk/include/reactos/ui/layout.h:
```diff
@@ -85,7 +85,7 @@ LayoutShowGrip(LAYOUT_DATA *pData, BOOL bShow)
     if (pData->m_hwndGrip == NULL)
     {
-        DWORD style = WS_CHILD | WS_CLIPSIBLINGS | SBS_SIZEGRIP;
+        DWORD style = WS_GROUP | WS_CHILD | WS_CLIPSIBLINGS | SBS_SIZEGRIP;
```
Also, this PR initially targeted [USER32] module, which is not the case anymore.

Gathub22 commented:
> Please undo all formatting/whitespace changes before merging.

I'm sorry for all this delay. Before I commited the new patch, I reverted both old commits. Surprisingly, the spaces my editor deleted by accident weren't detected by Git when I compared the reverted file with the master's one, and it was saying they were both the same. But after @binarymaster commit the old formatting is finally back.

Gathub22 commented:
I guess this PR is fixed and done. It should be merged

binarymaster commented:
@Gathub22 two unresolved comments: https://github.com/reactos/reactos/pull/6914#discussion_r1613922055 (and one after it)

Gathub22 commented:
Im really sorry for this long delay. Both comments have been resolved

HBelusca commented on sdk/include/reactos/ui/layout.h:
```diff
@@ -85,7 +85,7 @@ LayoutShowGrip(LAYOUT_DATA *pData, BOOL bShow)
     if (pData->m_hwndGrip == NULL)
     {
-        DWORD style = WS_CHILD | WS_CLIPSIBLINGS | SBS_SIZEGRIP;
+        DWORD style = WS_GROUP | WS_CHILD | WS_CLIPSIBLINGS | SBS_SIZEGRIP;
```
```suggestion
        /* CORE-19585: WS_GROUP set to avoid navigation over the grip control */
        DWORD style = WS_GROUP | WS_CHILD | WS_CLIPSIBLINGS | SBS_SIZEGRIP;
```

HBelusca reviewed (approved):
OK let's go with this change, for the time being.
My last suggestion is to add the comment in code to signal why the WS_GROUP has been added. (Because I still think we are kind of hackfixing the problem here.)
